### DISCUSSION TITLE
How to ignore major NPM updates

### DISCUSSION THREAD

**Discussion by tomkerkhove** [renovatebot/renovate#10707] [Request Help] [answered]
We are looking to ignore all NPM major updates but I can't get it to work. (I'm not an NPM user myself)

We currently use the following config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "branchPrefix": "renovate/",
  "azureAutoComplete": false,
  "labels": [
    "dependencies"
  ],
  "commitBodyTable": true,
  "packageRules": [
    {
      "packagePatterns": [
        "*"
      ],
      "groupName": "Others",
      "labels": [  ]
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchFiles": ["package.json"],
      "groupName": "NPM",
      "labels": [
        "NPM"
      ]
    }
  ],
  "extends": [
    "config:base",
    ":gitSignOff",
  ]
}
```

And since it's a last-one-wins approach it should only update NPM packages with minor or patch updates, but still we get PRs with major updates.

Are we missing something? Is there a preset for this?

_Sorry for the stupid question_

**Comment by rarkins** [comment, marked as answer]
Renovate updates everything by default, so you need a rule which *disables* major updates. Similar to what you have but with updateTypes=major and enabled=false. I'd recommend instead dependency dashboard approval instead of disabling, so you don't lose visibility of them

**Reply by tomkerkhove** [reply]
Makes sense, I will give that a try.

> I'd recommend instead dependency dashboard approval instead of disabling, so you don't lose visibility of them

Can you elaborate on this please?

**Reply by HonkingGoose** [reply]
The idea is to put a rule that says: "For all major npm updates, ask for approval via the DependencyDashboard first".

Put this new section within the `packageRules` array:

```json
    {
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true,
      "matchFiles": ["package.json"],
      "labels": [
        "NPM"
      ]
    }
```

I recommend turning on the Dependency Dashboard in the config as well, so it's always available.

Full config you have now, with the new rule + explicit Dependency Dashboard turned on:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "dependencyDashboard" : true,
  "branchPrefix": "renovate/",
  "azureAutoComplete": false,
  "labels": [
    "dependencies"
  ],
  "commitBodyTable": true,
  "packageRules": [
    {
      "packagePatterns": [
        "*"
      ],
      "groupName": "Others",
      "labels": [  ]
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchFiles": ["package.json"],
      "groupName": "NPM",
      "labels": [
        "NPM"
      ]
    },
    {
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true,
      "matchFiles": ["package.json"],
      "labels": [
        "NPM"
      ]
    }
  ],
  "extends": [
    "config:base",
    ":gitSignOff",
  ]
}
```

**Reply by viceice** [reply]
@HonkingGoose i would prefer to not group major updates. So they can be approved one by one

**Reply by HonkingGoose** [reply]
@viceice I've updated my post. 😄

**Reply by HonkingGoose** [reply]
Maybe this is even better?

```json
{
  "packageRules": [
    {
      "matchManagers": ["npm"],
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true,
    }
  ]
}
```

**Reply by tomkerkhove** [reply]
It looks like this is not supported for Azure DevOps though:
> WARN: ensureIssue() is not implemented
> ...
> WARN: findIssue() is not implemented

But that's ok - Thank you!

**Reply by tomkerkhove** [reply]
https://github.com/renovatebot/renovate/issues/10725
